lib/logstash/inputs/snmp/client.rb

```diff
@@ -136,8 +136,14 @@ def build_target(address, community, version, retries, timeout)
     end
 
     def parse_version(version)
-      # TODO implement
-      SnmpConstants.version2c
+      case version.to_s
```

Comment:
it's starting to become evident that we're doing parameter validation twice, once by validating the uri contains valid elements, and then revalidating when creating the snmp client, perhaps the regexes could be removed?

Reply:
I'd like to keep it like this because:
- in the register method I can throw a `ConfigurationError`, this is where it belongs. Ideally I'd even prefer to have a custom validator so that the validation happen before register, like for other config.
- the `Client` class manages its own concerns, and does also have to do parameter validation so it's ok to actually have it at both places IMO.

Reply:
Also note that I will rework the parameter validation bits in the near future so I don't think this is too important for now, as long as its working as intended.